fastlane/lib/fastlane/helper/sh_helper.rb

```diff
@@ -52,7 +59,7 @@ def self.sh_control_output(command, print_command: true, print_command_output: t
           message += "\n#{result}" if print_command_output
 
           error_callback.call(result) if error_callback
-          UI.shell_error!(message)
+          UI.shell_error!(message) if raise_shell_error
```

Comment:
You're right.
I didn't think deeply.

I'll use instead `UI.error(message)` when this options is disabled.